satellite/contact/endpoint.go

```diff
@@ -37,46 +39,69 @@ func NewEndpoint(log *zap.Logger, service *Service) *Endpoint {
 func (endpoint *Endpoint) Checkin(ctx context.Context, req *pb.CheckinRequest) (_ *pb.CheckinResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	pingNodeSuccess, pingErrorMessage, err := pingBack(ctx, endpoint, req)
+	peerID, err := peerIDFromContext(ctx)
+	pingNodeSuccess, pingErrorMessage, err := pingBack(ctx, endpoint, req, peerID)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
 
-	// Update the overlay cache with new uptime info from pinging back the node
-	peerID, err := peerIDFromContext(ctx)
+	// TODO(jg): We are making 2 requests to the database, one to update uptime and
+	// the other to update the capacity and operator info. We should combine these into
+	// one to reduce db connections. Consider adding batching and using a stored procedure.
 	_, err = endpoint.service.overlay.UpdateUptime(ctx, peerID, pingNodeSuccess)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
 
-	// Should we update the req.NodeCapacity here as well?
+	nodeInfo := pb.InfoResponse{Operator: req.GetOperator(), Capacity: req.GetCapacity()}
+	_, err = endpoint.service.overlay.UpdateNodeInfo(ctx, peerID, &nodeInfo)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
 
 	return &pb.CheckinResponse{
 		PingNodeSuccess:  pingNodeSuccess,
 		PingErrorMessage: pingErrorMessage,
 	}, nil
 }
 
-func pingBack(ctx context.Context, endpoint *Endpoint, req *pb.CheckinRequest) (bool, string, error) {
+func pingBack(ctx context.Context, endpoint *Endpoint, req *pb.CheckinRequest, peerIDFromContext storj.NodeID) (bool, string, error) {
 	client, err := newClient(ctx,
 		endpoint.log,
 		endpoint.service.transport,
 		req.GetAddress(),
 	)
-	defer func() {
-		err = errs.Combine(err, client.close())
-	}()
 	if err != nil {
-		return false, "", err
+		// if this is a network error, then return the error otherwise just report internal error
+		_, ok := err.(net.Error)
+		if ok {
+			return false, "", Error.New("couldn't connect to client at addr %s due to network error: %v.", req.GetAddress().String(), err)
```

Comment:
The "client" term might be confusing here, since normally we refer to the destination of a connect attempt as a server. I mean, I get it, the originating node is a client and we're trying to connect back to it, but people might see this error out of context. How about something more concise like `"failed to connect to %s: %v"`?